project.mml

```diff
@@ -611,8 +611,14 @@ Layer:
       <<: *osm2pgsql
       table: &turning-circle_sql |-
         (SELECT DISTINCT ON (p.way)
-            p.way AS way,
+            -- Valid line geometry requires at least two points that are NOT identical, therefore moving them slightly.
+            ST_MakeLine(ST_Translate(p.way, -0.0001, -0.0001), ST_Translate(p.way, 0.0001, 0.0001)) AS way,
```

Comment:
I don't like generating fake lines. Why can't we use a point and crop out an appropriately sized circle from the unpaved pattern?

Reply:
That would mean cutting about 30 different circular SVGs from the patterns.  And you would still get a discontinuity of the pattern rendering at the transit from the turning circle to the linear road.

The other alternative would be using ST_Buffer() + polygon pattern, but that would require the road width progression to be available in SQL. On the other hand, that would allow merging the turning circle and highway area layers and this way could reduce size of roads.mss quite a bit...

Reply:
As @imago explained, using a point geometry

> would mean cutting about 30 different circular SVGs from the patterns

Therefore, I think what is proposed in this PR is easier in code and maintenance.